relay-cardinality/src/redis/quota.rs

```diff
@@ -74,7 +98,37 @@ impl QuotaScoping {
         let organization_id = self.organization_id.unwrap_or(0);
         let project_id = self.project_id.map(|p| p.value()).unwrap_or(0);
         let namespace = self.namespace.map(|ns| ns.as_str()).unwrap_or("");
+        let name = DisplayOptMinus(self.name);
+
+        format!("{KEY_PREFIX}:{KEY_VERSION}:scope-{{{organization_id}-{project_id}-{namespace}}}-{name}{slot}")
+    }
+}
+
+impl std::ops::Deref for QuotaScoping {
+    type Target = PartialQuotaScoping;
+
+    fn deref(&self) -> &Self::Target {
+        &self.parent
+    }
+}
+
+struct DisplayOptMinus<T>(Option<T>);
```

Comment:
Is the goal of this struct to display the inner object optionally? If yes, I find the extra complexity is not needed and I'd suggest moving the logic to line 101 above or defaulting to a fall-back value instead.

Reply:
We have the same in `relay-quota`, we need it to stay compatible with the existing keys.